make/lint.mk

```diff
@@ -10,6 +10,10 @@ lint-yaml: ${YAML_FILES}
 
 .PHONY: lint-go-code
 ## Checks the code with golangci-lint
-lint-go-code: generate
-	$(Q)go get github.com/golangci/golangci-lint/cmd/golangci-lint
-	$(Q)${GOPATH}/bin/golangci-lint ${V_FLAG} run
\ No newline at end of file
+lint-go-code: generate install-golangci-lint
+	$(Q)${GOPATH}/bin/golangci-lint ${V_FLAG} run -E gofmt,golint,megacheck,misspell ./...
+
+.PHONY: install-golangci-lint
+## Install development tools.
+install-golangci-lint:
+	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.39.0
```

Comment:
We should be very careful with running any scripts from master branches from some github repos we can't 100% trust. It's not safe.

Reply:
yeah, we should fix that also in our sandbox-sre repo

Reply:
I understand your concern. In sandbox-sre, we use the GitHub action, and this script is only used locally.
There are a few alternatives to installing and running locally, via `brew` on mac or via a Docker image: https://golangci-lint.run/usage/install/
I'm fine with discussing what's the best solution for us. In particular, I was wondering if we could start using the GH action for all our repositories, instead of OpenShift CI jobs? WDYT?

Reply:
Yes, I think we can/should move all jobs but e2e tests to GitHub Actions.
And yes, I noticed that the linter is used locally only. We still should not run any scripts from 3rd party master branches in our make targets.
Let's remove it from this PR and commit the fixes found by the linter only. We can discuss our options with installing the linter separatly.

Reply:
reverted in e801db8, and we can discuss about next steps later